### ISSUE TITLE
[samsungtv] Huge amount of threads over time

### ISSUE THREAD

**Issue by Matsuo3rd** [openhab/openhab-addons#12534]
## Expected Behavior
openHab's JVM shall have "stable" number of threads.

## Current Behavior

openHab's JVM holds +1000 threads after 5 days running.
Running `jstack <JVM_PID>` returns all threads hold by JVM.
Looking at the ouput, and searching for pattern `OH-httpClient-samsungtv-` returns oddly high amount of threads (+700).
Conclusion: samsungtv plugin does not properly manage threads lifecycle.

## Your Environment
1. Running openHab stable 3.2.0
2.Running samsungtv binding included in openHab stable 3.2.0
3. 2014 Samsung TV (binding only used for ON/OFF detection). Items works as advertized.

.things file
`Thing samsungtv:tv:11490c82_0022_1000_ab77_380195fef270 "Samsung TV LED48" [ hostName="192.168.1.45", protocol="None", macAddress="38:01:95:7D:8D:74", refreshInterval=10000 ]`

.items file
`Switch samsungtv_power	"Samsung TV State"	{channel="samsungtv:tv:11490c82_0022_1000_ab77_380195fef270:power"}`

**Comment by Matsuo3rd** [comment]
Behavior still observed with [org.openhab.binding.samsungtv-3.3.0-SNAPSHOT.jar](https://ci.openhab.org/job/openHAB-Addons/lastSuccessfulBuild/artifact/bundles/org.openhab.binding.samsungtv/target/org.openhab.binding.samsungtv-3.3.0-SNAPSHOT.jar)

**Comment by jlaur** [comment]
@Matsuo3rd - FWIW, I don't have this behavior with my 2011 Samsung TV using legacy protocol. So might be related to WebSocket implementation for newer protocols. There is an active PR with major changes, #11895. @NickWaterton, is this something that you have become aware of while working on that PR?

**Comment by NickWaterton** [comment]
Yes,

I can't remember exactly, but I don't think the original binding was using the http pool, and was using thread:sleep for timing.

I switched things to use pooled resources and the scheduler for timing to cut down on resource use in the revised binding.

You can try the revised binding, and see if it cut down the threads usage.

**Comment by Matsuo3rd** [comment]
Hello,

I actually removed Samsung binding and switched to Network binding in order to cover my need (detect TV state ON/OFF: a simple ping does the job).

When I was investigating threads issues with Samsung binding I looked at the logs and found out that my initial files based configuration was in some way overridden by UPnP auto discovery.
ie: I initially setup a "None" protocol (no need to control TV) but eventually TV gets automatically rediscovered and gets assigned a "WebSocket" protocol. Then errors are triggered as binding cannot connect to TV using websocket (I don't even know if my TV supports it).
I suspect (no proof) that threads issues is coming from that.

See logs attached (samsung.log.3)
[samsungtv.log.3.zip](https://github.com/openhab/openhab-addons/files/8590240/samsungtv.log.3.zip)

line 345:  last reference of "Using None interface" (as configured in files)
then
line 349: remoteDeviceAdded, protocol: WebSocket

Hope it helps.
